src/utils/format_json/format_json.c

```diff
@@ -397,6 +404,28 @@ static int format_json_value_list_nocheck(char *buffer, /* {{{ */
   return 0;
 } /* }}} int format_json_value_list_nocheck */
 
+static int format_json_meta_data_nocheck(char *buffer, /* {{{ */
+                                         size_t *ret_buffer_fill,
+                                         size_t *ret_buffer_free,
+                                         meta_data_t *meta, size_t temp_size,
+                                         char **metadata_keys,
+                                         size_t metadata_keys_num) {
+  char temp[temp_size];
+  int status;
+
+  status = meta_data_to_json(temp, sizeof(temp), meta, metadata_keys,
+                             metadata_keys_num);
+  if (status != 0)
+    return status;
+  temp_size = strlen(temp);
+
+  memcpy(buffer + (*ret_buffer_fill), temp, temp_size + 1);
```

Comment:
Reads one item past `temp` array end which is (eventually) going to trigger compiler warnings (and fail builds with `-Werror`).

Looking at existing code, `format_json_value_list_nocheck()` seems to have the same bug.

Reply:
Ah, although it will be guarded just above `if (status != 0)` (`-ENOMEM`) in actual, certainly it's not good.
Probably we should use another variable and check length before calling `memcpy()`.

Reply:
No, it's OK after all, I'm just blind. I somehow managed to missed the strlen() call. :-/